frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/ticketGroupsContext.component.tsx

```diff
@@ -16,147 +16,96 @@
 }: TicketGroupsContextComponentProps) => {
-	const [indexedOverrides, setIndexedOverrides] = useState(addIndex(overrides));
-	const [groupsTree, setGroupTree] = useState<GroupNode>(null);
-	const [selectedIndexes, setSelectedIndexes] = useState<number[]>([]);
+	// overrides should arrive already indexed
+	const [indexedOverrides, setIndexedOverrides] = useState<IndexedOverride[]>(_.sortBy(addIndex(overrides), 'prefix'));
+	const [selectedIndexes, setSelectedIndexes] = useState<Set<number>>(new Set(indexedOverrides.map((o, i) => i)));
 
-	const getGroupNode = (prefix = []): GroupNode => {
-		if (!groupsTree) return null;
-		let currentNode = groupsTree;
-		prefix.forEach((prefixSegment) => {
-			currentNode = currentNode.children.find((c) => c.prefixSegment === prefixSegment);
-		});
-		return currentNode;
-	};
-
-	const getNodeParent = (index) => getGroupNode(indexedOverrides.find((g) => g.index === index).prefix);
-
-	const getGroupLeaf = (index) => getNodeParent(index).children.find((c) => c.index === index);
-
-	const getCollectionState = (prefix?) => getGroupNode(prefix)?.state ?? null;
-
-	const backpropagateNodeUpdate = (node: GroupNode) => {
-		if (node.children.length) {
-			const firstChildState = node.children[0].state;
-			const allChildrenStatesAreEqual = node.children.every((c) => c.state === firstChildState);
-			const newState = allChildrenStatesAreEqual ? firstChildState : GroupState.INDETERMINATE;
-			if (node.state === newState) return;
-			node.state = newState;
-		}
-		if (node.parent) {
-			backpropagateNodeUpdate(node.parent);
-		}
-	};
-
-	const updateTree = (currentNode) => {
-		backpropagateNodeUpdate(currentNode);
-		const newTree = { ...groupsTree };
-		// next line is necessary because after the line above the reference
-		// to the parent node in root's children does not match the root anymore
-		newTree.children.forEach((c) => { c.parent = newTree; });
-		setGroupTree(newTree);
-	};
-
-	const setGroupNodeState = (groupNode: GroupNode, newState: Exclude<GroupState, GroupState.INDETERMINATE>) => {
-		if (groupNode.state === newState) return;
-		groupNode.state = newState;
-		groupNode.children.forEach((c) => setGroupNodeState(c, newState));
-	};
+	const getGroupState = (index) => selectedIndexes.has(index);
 
 	const toggleGroupState = (index) => {
-		const groupLeaf = getGroupLeaf(index);
-		setGroupNodeState(groupLeaf, groupLeaf.state * (-1));
-		updateTree(groupLeaf);
-	};
-
-	const toggleCollectionState = (prefix = []) => {
-		const groupNode: GroupNode = getGroupNode(prefix);
-		const newState = groupNode.state === GroupState.CHECKED ? GroupState.UNCHECKED : GroupState.CHECKED;
-		groupNode.children.forEach((node) => setGroupNodeState(node, newState));
-		updateTree(groupNode);
-	};
-
-	const shiftIndex = (node: GroupNode, removedIndex: number) => {
-		if (node.index && node.index > removedIndex) {
-			node.index--;
-			return;
+		if (selectedIndexes.has(index)) {
+			selectedIndexes.delete(index);
```

Comment:
instead of `selectedIndexes` do you think `checkedIndexes` suits it better? Because you are using checked in the rest of the code it might be a more fitting wording.

Whichever you feel is better.